ui/app/components/app/transaction-time-remaining/transaction-time-remaining.container.js

```diff
@@ -0,0 +1,84 @@
+import { connect } from 'react-redux'
+import {
+    getEstimatedGasPrices,
+    getEstimatedGasTimes,
+    getCustomGasPrice,
+    formatTimeEstimate,
+    getFastPriceEstimateInHexWEI,
+} from '../../../selectors/custom-gas'
+import TransactionTimeRemaining from '../transaction-time-remaining/transaction-time-remaining.component'
+import { getAdjacentGasPrices, extrapolateY } from '../gas-customization/gas-price-chart/gas-price-chart.utils'
+import { hexWEIToDecGWEI } from '../../../helpers/utils/conversions.util'
+import { getSelectedToken } from '../../../selectors/selectors.js'
+import { fetchBasicGasAndTimeEstimates } from '../../../ducks/gas/gas.duck'
+
+const mapStateToProps = (state, ownProps) => {
+    const { selectedAddressTxList } = state.metamask
+    const { modalState: { props: modalProps } = {} } = state.appState.modal || {}
+    const { transaction = {} } = ownProps
+    const { txData = {} } = modalProps || {}
+    const selectedTransaction = selectedAddressTxList.find(({ id }) => id === (transaction.id || txData.id))
+    const { gasPrice: currentGasPrice } = getTxParams(state, selectedTransaction)
+    const customModalGasPriceInHex = getCustomGasPrice(state) || currentGasPrice
+    const customGasPrice = calcCustomGasPrice(customModalGasPriceInHex)
+    const gasPrices = getEstimatedGasPrices(state)
+    const estimatedTimes = getEstimatedGasTimes(state)
+
+    return {
+        customGasPrice,
+        currentTimeEstimate: getRenderableTimeEstimate(customGasPrice, gasPrices, estimatedTimes),
```

Comment:
I also wonder if performance could be improved by ensuring that `getRenderableTimeEstimate` is called only once for each transaction.

Reply:
I will refeactor `getRenderableTimeEstimate` to a higher component to see how it impacts on performance.